qiskit_experiments/base_experiment.py

```diff
@@ -107,6 +106,8 @@ def run(
         if experiment_data is None:
             # Create new experiment data
             experiment_data = self.__experiment_data__(experiment=self, backend=backend)
+            experiment_data.metadata()["experiment_data"] = self._type
```

Comment:
@chriseclectic Can you please check this one?  See also #124.

Reply:
This isn't necessary since there is already the `experiment_data.experiment_type` property

Reply:
This shouldn't be needed since it is already added as part of experiment `_metadata` method which is called by ExperimentData constructor (though in light of above comment probably shouldn't even be in metadata since its already a property of ExperimentData)